Fix aten.amax lowering issue

cccclai opened the pull request:
Summary:
There was an error when lowering amax around this line `input_tensor = self.get_tensor(input_node, node)` and the issue is that we're trying to permute the tensor inside node_visitors, op_node.meta[QCOM_AXIS_ORDER] is (0, 1), however, tensor.shape is (1, 980, 49).

Rollback Plan:

Differential Revision: D80187368

cccclai commented:
This PR fix the lowering issue, however, not quite sure a good unit test for it. Try to use the following code but looks like passing without this PR. Do you know what might trigger the error?

```
class ConvAMax2D(torch.nn.Module):
    def __init__(self, in_channels, out_channels, kernel_size=3, dim=None, keepdim=False):
        super().__init__()
        self.conv = torch.nn.Conv2d(in_channels, out_channels, kernel_size, padding=kernel_size//2)  
        self.dim = dim
        self.keepdim = keepdim

    def forward(self, x):
        x = self.conv(x) 
        return torch.amax(x, dim=self.dim, keepdim=self.keepdim)

sample_input = torch.randn(2, 3, 64, 64) 
module = ConvAMax2D(in_channels=3, out_channels=16, kernel_size=3, dim=1, keepdim=False)
output = module(sample_input)
```

cccclai commented:
Can I get a review on this PR? @haowhsu-quic @shewu-quic @winskuo-quic @DannyYuyang-quic

haowhsu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -1435,6 +1442,13 @@ def test_qnn_backend_amax(self):
+        module = AMaxFollowingConv2D(  # noqa: F405
+            in_channels=3, out_channels=16, kernel_size=3, dim=-1, keepdim=False
+        )
+        self.lower_module_and_test_output(module, sample_input)
```
I think we should have `module = self.get_qdq_module(module, sample_input)` here.

cccclai replied:
updated

cccclai committed:
Fix aten.amax lowering issue (#13381)

Summary:

There was an error when lowering amax around this line `input_tensor = self.get_tensor(input_node, node)` and the issue is that we're trying to permute the tensor inside node_visitors, op_node.meta[QCOM_AXIS_ORDER] is (0, 1), however, tensor.shape is (1, 980, 49).

Rollback Plan:

Differential Revision: D80187368
